src/ScriptBuilder/Meters/MeterDefinitionReader.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace NServiceBus.Metrics.PerformanceCounters
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using Mono.Cecil;
+
+    static class MeterDefinitionReader
+    {
+        public static bool TryGetMeterDefinition(TypeDefinition type, out List<MeterDefinition> definitions)
+        {
+            definitions = new List<MeterDefinition>();
+            if (type.BaseType != null && type.BaseType.FullName == "NServiceBus.Metrics.MetricBuilder")
+            {
+                var meterAttributes = type.Fields.Select(f => f.GetSingleAttribute("NServiceBus.Metrics.MeterAttribute")).Where(c => c != null);
+
+                foreach (var meterAttribute in meterAttributes)
+                {
+                    var name = (string)meterAttribute.ConstructorArguments[0].Value;
```

Comment:
@SimonCropp do you know of any better way of reading property value from CustomAttribute? `Properties` and `Fields` collection is empty. The MeterAttribute class has properties (with getters only). During runtime (while debugging) when Iw ill write meterAttribute.AttributeType.Properties it returns filled the collection. As during runtime `AttributeType` returns type `MeterAttribute`. But during compile time the AttributeType return CustomAttribute :|